torchbenchmark/util/framework/timm/model_factory.py

```diff
@@ -81,7 +81,10 @@ def enable_channels_last(self):
         self.example_inputs = self.example_inputs.contiguous(memory_format=torch.channels_last)
 
     def enable_amp(self):
-        self.amp_context = lambda: torch.cuda.amp.autocast(dtype=torch.float16)
+        if self.device == "cuda":
+            self.amp_context = lambda: torch.cuda.amp.autocast(dtype=torch.float16)
```

Comment:
Let's use `self.amp_autocast = torch.cuda.amp.autocast` here.

Reply:
Do you mean `self.amp_context` here?